examples/put_to_self.c

```diff
@@ -64,73 +65,94 @@ int main(void)
 	}
 
 	ret = PtlNIInit(PTL_IFACE_DEFAULT, PTL_NI_NO_MATCHING | PTL_NI_PHYSICAL, PTL_PID_ANY, NULL,
-			NULL, &retNIInit);
+			NULL, &nih);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto fini;
 	}
 
-	ret = PtlEQAlloc(retNIInit, 10, &retEQAlloc);
+	ret = PtlEQAlloc(nih, 10, &eqh);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto ni_fini;
 	}
 
-	ret = PtlPTAlloc(retNIInit, PTL_PT_FLOWCTRL, retEQAlloc, PTL_PT_ANY, &retPTAlloc);
+	ret = PtlPTAlloc(nih, PTL_PT_FLOWCTRL, eqh, PTL_PT_ANY, &pti);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto eq_free;
 	}
 
-	ret = PtlGetId(retNIInit, &id);
+	ret = PtlGetId(nih, &id);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
 		return 1;
 	}
 
-	/*prepare our events buffer to receive a message represented by le*/
-	ret = PtlLEAppend(retNIInit, retPTAlloc, &le, PTL_PRIORITY_LIST, NULL, &retAppend);
+	/* expose to the network a portion of the process address space */
+	ret = PtlLEAppend(nih, pti, &le, PTL_PRIORITY_LIST, NULL, &leh);

```

Comment:
I would prefer to have the le structure filled before the call to PtlLEAppend, rather than during variable initialization.
same for md structure filling.